### PR TITLE
YouTube new video notification feature.

### PR DESCRIPTION
**Pull request by Leon2ev**
We can store crypto-related channels inside the 'youtubeChannels.json' file. Every 1 min program will loop through this list of channels and check if there are any new videos out. If a new video is out we will get a message to our Discord channel. As Phero suggested, we should create a promote-youtube channel, so all the links will go only into these channels and our leaders can have access to newly published videos. When the 'promote youtube channel will be created we should add this channel ID to the rest channel ID, I have created property as 'promoteOnYouTube'.

### CODE REVIEW

File: src/tasks/promote/you-tube/notifications.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  getChannels(): void {
+    try {
+      const data = fs.readFileSync(path.resolve(__dirname, '../../youtubeChannels.json'), 'utf8');
+      const { channels } = JSON.parse(data);
+      this.youtubeChannels = channels;
+    } catch(e) {
+      console.error('Cannot read youtubeChannels.json:', e);
+    }
+  }
+
```
Comment (tranpeter08) [inline src/tasks/promote/you-tube/notifications.ts:66]: It is not set in our `tsconfig.json` file yet, but adding `"resolveJsonModule": true` will allow us to just import the file like so: 
```javascript
import youTubeChannels from '../../youtubeChannels.json''` 
```

Comment (Leon2ev) [reply src/tasks/promote/you-tube/notifications.ts:66]: Ok. Will fix that.

**Comment by tranpeter08** [comment]
I added my YouTube channel to the list and uploaded a couple of videos.  Sometimes it would work as expected and the video would show up in the feed shortly after upload was completed.  Then there were times that it took up to 30 minutes for the new video to show up in the feed, and because the time that elapsed between the publish date and the time the feed was read was greater than 5 minutes, the bot didn't send out a message.

**Comment by Leon2ev** [comment]
So basically my script is working? The problem is that the video is not showing up in feed instantly right?. Maybe we can just set 1 hour instead of 5 minutes in this case we can remove feed update issue. What do you think?

**Comment by tranpeter08** [comment]
> So basically my script is working? The problem is that the video is not showing up in feed instantly right?. Maybe we can just set 1 hour instead of 5 minutes in this case we can remove feed update issue. What do you think?

I'm sorry for the late responses. I work 60-80 hours a week for my regular job.  

Yes, your script is perfectly fine. Just need to figure why the vid isn't being added to the feed right away. I feel that if we set the period to 1 hour we might miss some videos. For example, what if a YouTuber creates a short and uploads a regular video around the same time?

**Comment by Leon2ev** [comment]
Yes, I understand your point. Everyone is busy and we have a huge time difference as well. Looks like I found an even better solution, straight to the Discord. I will implement it later today. 
https://npm.io/package/discord-youtube-notifications

**Comment by Leon2ev** [comment]
> Yes, I understand your point. Everyone is busy and we have a huge time difference as well. Looks like I found an even better solution, straight to the Discord. I will implement it later today.
> https://npm.io/package/discord-youtube-notifications

I did check both libraries, they are broken and there is no source code available. I will update my code to make it work. Maybe I can make my own package for YouTube afterward.

**Pull request closed without merging**